docs/rules/strict-component-boundaries.md

```diff
@@ -0,0 +1,25 @@
+# Prevent module imports between components. (strict-component-boundaries)
+
+Components should have everything they need co-located in a single directory. This makes it easy to move components to a more appropriate spot, and makes it easy for future developers to find, understand, and update everything about a component in one place.
```

Comment:
I think need a comment that other components should not be reaching in/ the component should export its entire API